libs/framework/include/celix_bundle_context.h

```diff
@@ -126,8 +126,38 @@ typedef struct celix_service_registration_options {
      * for this.
      */
     const char *serviceVersion OPTS_INIT;
+
+    /**
+     * If set to > 0, this svc id will be used to register the service.
+     * The reservedSvcId should be reserved with a call to celix_bundleContext_reserveSvcId
+     */
+    long reservedSvcId OPTS_INIT;
 } celix_service_registration_options_t;
 
+/**
+ * Reserves a service id, which is expected to be used to register a service in the future.
+ *
+ * If a celix_bundleContext_unregisterService with the reserved service id is called earlier than
+ * the celix_bundleContext_registerServiceWithOptions with the reserved service, the registration will be be cancelled.
+ * This means then when the expected celix_bundleContext_registerServiceWithOptions call happens this will not
+ * result in a service registration and a  -2 will be returned as service id.
+ *
+ * Subsequent calls to celix_bundleContext_registerServiceWithOptions for an already cancelled service id will
+ * return a -1 error code (and a 'Invalid reservedSvcId' error log entry will be logged).
```

Comment:
Shouldn't that be a -2?

Reply:
no,  Registering with a reserved id which is unknown gives a -1. Whether this is due to using an already cancelled reserved service id or "just" a invalid reserved service id is not tracked.

So the bundle context keeps track of reserved service ids for future service registrations (to follow through with the registration or when cancelled skip the registration), but after that they are not tracked any more.

Reply:
To me, this is confusing, so only the first registration will trigger a cancelled. That makes it unclear/vague for users to know what is going on if someone does a second registration for whatever reason.